tensorrt_llm/_torch/speculative/ngram.py

```
@@ -198,3 +198,6 @@ def prepare_draft_tokens(
                 pad_length = self.max_draft_len - len(draft_tokens)
                 draft_tokens.extend([request.py_end_id] * pad_length)
             request.py_draft_tokens = draft_tokens
+
+    def should_use_spec_decode(self, requests: List[LlmRequest]) -> bool:
```

Ruff (0.12.2)

`List` may be undefined, or defined from star imports

(F405)